### PR TITLE
[INFRA-99, INFRA-100]: Feat/delete unreferenced files (from users & backups)

### PR DESCRIPTION
**Pull request by fabioespinosa**
Iterates over users, then over BucketEntries to find a file that is not in the files table of drive. If so, deletes it by sending a request to bridge.

### CODE REVIEW

File: bin/delete-unreferenced-files.js [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+) => {
+  const idFile = entry._id.toString();
+
+  sqlPool.query(sqlFilesQuery, [idFile], (err, results) => {
+    if (err) {
+      return cb(err);
+    }
+
+    const count = results[0].count;
+    if (count === 0) {
```
Comment (sg-gs) [inline bin/delete-unreferenced-files.js]: Reverse the if to and add a return to make the code more readable:

```
if (count > 0) {
  return cb();
}
...
```

**Review by sg-gs — CHANGES_REQUESTED** [review]
As this file is big, could be a better idea to change delete-unreferenced-files.js to a folder with HTTP methods and SQL queries in small files while having an entry point for the command. P. e:

```
 /delete-unreferenced-files
   /index.js -> where the command is
   /requests.js -> where HTTP requests are
   /database.js -> where database connections and queries are made
```

File: bin/delete-unreferenced-files.js (hunk 1 above)
Comment (fabioespinosa) [reply bin/delete-unreferenced-files.js]: Fixed

**Comment by fabioespinosa** [comment]
> As this file is big, could be a better idea to change delete-unreferenced-files.js to a folder with HTTP methods and SQL queries in small files while having an entry point for the command. P. e:
> 
> ```
>  /delete-unreferenced-files
>    /index.js -> where the command is
>    /requests.js -> where HTTP requests are
>    /database.js -> where database connections and queries are made
> ```

refactored into database.js, request.js and index.js

File: bin/cleaner/database.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  each(entries, onEveryEntry, cb);
+};
+
+function iterateOverModel(Model, filter, onEveryEntry, cb) {
+  let chunk = [];
+  const chunkSize = 5;
+
+  const cursor = Model.find(filter)
```
Comment (sg-gs) [inline bin/cleaner/database.js]: Better inject the `cursor` directly. As the cursor uses the EventEmitter API, could be replaced for another EventEmitter implementer, also you could avoid coupling things like sorting inside this function. At the same time, you could gain flexibility, allowing the query to be a find with any condition, etc without changing this function

Comment (fabioespinosa) [reply bin/cleaner/database.js]: Agree, let's change it to:

`iterateOverModel -> iterateOverCursor`

**Pull request merged**